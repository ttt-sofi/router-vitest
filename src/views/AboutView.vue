<template>
    <div class="about">
        <h1>Vista de About</h1>
        <p>Esta es la vista de sobre</p>
    </div>
</template>

<script>
</script>