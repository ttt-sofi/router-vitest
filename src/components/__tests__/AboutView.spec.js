import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import AboutView from "@/views/AboutView.vue";
import {describe,test, expect} from "vitest";

describe("AboutView", () => {
  test("Probando la existencia del componente o vista AboutView ", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/about",
          name: "AboutViewVue",
          component: AboutView,
        },
      ],
    });
    router.push("/about");
    await router.isReady();
    const wrapper = mount(AboutView, {
      global: {
        plugins: [router],
      },
    });
    expect(wrapper.findComponent(AboutView).exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("DeberÃ­a contener el texto 'Vista de About'", () => {
    const wrapper = mount(AboutView);
    expect(wrapper.text()).toContain("Vista de About");
  });
});