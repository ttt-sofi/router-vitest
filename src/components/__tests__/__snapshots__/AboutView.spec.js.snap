// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutView > Probando la existencia del componente o vista AboutView  1`] = `
"<div class="about">
  <h1>Vista de About</h1>
  <p>Esta es la vista de sobre</p>
</div>"
`;
